### DISCUSSION TITLE
Quick way to concat/merge all files from directory?

### DISCUSSION THREAD

**Discussion by bwiggs** [saulpw/visidata#2593] [Q&A] [answered]
Say I have a directory full of csv or json files, they all have the same schema. What's the most efficient way to open and view all of them in a single sheet? For my actual use case, I have a directory which contains a bunch of json files each a single object.

From what I can tell the current way to do this is to `vd *.json` then go to the sheets sheet, select all the files, `&` and `concat`? It would be really neat if there was a fast way to load all those files as a single sheet with a simple option to `vd`. Perhaps I missed it in the docs?

**Comment by saulpw** [comment, marked as answer]
Here's a script (untested) that concatenates json files:

```
import sys
import json

objs = []
for fn in sys.argv[1:]:
    objs.append(json.loads(open(fn).read()))
json.dumps(objs)
```

If script is called "jsoncat.py" then you can do:

`jsoncat.py *.json | vd`

**Reply by bwiggs** [reply]
OK, thanks Saul! Makes sense.
